### ISSUE TITLE
About the preprocessed superpoints from Box2Mask when preprocessing S3DIS.

### ISSUE THREAD

**Issue by LinLin1031** [VinAIResearch/ISBNet#40]
Hello! If I want to work with a custom dataset similar to S3DIS, how should I get the preprocessed superpoints from Box2Mask? @ngoductuanlhp @datquocnguyen

**Comment by ngoductuanlhp** [comment]
Hi @LinLin1031,
You could try these repos [https://github.com/fpthink/SPNet](https://github.com/fpthink/SPNet) and [https://github.com/loicland/superpoint_graph](https://github.com/loicland/superpoint_graph).

Best.

**Comment by LinLin1031** [comment]
Could you please be a little more specific? @ngoductuanlhp

**Comment by kaoutar-elmouh** [comment]
> Bonjour! Si je souhaite travailler avec un ensemble de données personnalisé similaire à S3DIS, comment dois-je obtenir les superpoints prétraités de Box2Mask ?@ngoductuanlhp @datquocnguyen

I have the same question if someone can answer us

**Comment by Jack-Dong-Dong** [comment]
I'm also planning to use the algorithm on a custom dataset, but I'm having problems with the “superpoint”, have you solved it yet?

**Comment by Jack-Dong-Dong** [comment]
I found method in other community to generate superpoint using “segmentator.segment_mesh(vertices, faces).numpy()”, the method to generate mesh is provided in open3d but I don't understand it a bit! The format of the raw data, do I need to generate the mesh for each different label and then merge them together or just use the whole scene to generate it? Looking forward to your reply!@ngoductuanlhp

**Comment by yyds12** [comment]
Hello, have you solved how to generate preprocessing superpoints files for S3DIS format data? @LinLin1031 @kaoutar-elmouh @Jack-Dong-Dong

**Comment by Jack-Dong-Dong** [comment]
> Hello, have you solved how to generate preprocessing superpoints files for S3DIS format data? @LinLin1031 @kaoutar-elmouh @Jack-Dong-Dong
I use the data provided in the original text：
https://datasets.d2.mpi-inf.mpg.de/box2mask/segment_labels.tar.gz

**Comment by yyds12** [comment]
Yes, it's also mentioned in Box2Mask, except that repo also just gives a download link for oversegmentation, but it's not clear how this is generated. If you switch to your own dataset, you still don't know how to generate the corresponding file for that pre-processing. Examining the .npy file under the file. There are five values for 'segment', 'per_point_segment_instanceID', 'seg2instanceID', 'per_point_segment_semanticID', 'seg2semanticID', and 'seg2semanticID'. attribute information. The headache is that there is no corresponding xyz coordinates for each point, so this can't be realized in the usual way. @Jack-Dong-Dong

**Comment by Jack-Dong-Dong** [comment]
> Yes, it's also mentioned in Box2Mask, except that repo also just gives a download link for oversegmentation, but it's not clear how this is generated. If you switch to your own dataset, you still don't know how to generate the corresponding file for that pre-processing. Examining the .npy file under the file. There are five values for 'segment', 'per_point_segment_instanceID', 'seg2instanceID', 'per_point_segment_semanticID', 'seg2semanticID', and 'seg2semanticID'. attribute information. The headache is that there is no corresponding xyz coordinates for each point, so this can't be realized in the usual way. @Jack-Dong-Dong

I found a solution on the blog, and it ran successfully: to generate superpoints index according to the segment_mesh method provided by the segmentator package, you need to pass two parameters, which are the parameters of TriangleMesh, for specific introduction, you can refer to the content related to TriangleMesh in open3d, and TriangleMesh is also generated according to the method provided in open3d, I understand this superpoint as a triangular surface to describe the point cloud features, considering that the data is large, it can be generated based on a single instance and combined together, the above is my approach, I hope to help you. There's an option in the config file to use superpoint, and I think it's possible to change it to false, maybe ignoring this

**Comment by yyds12** [comment]
Thank you very much for your response; your answer has been very helpful to my research. I’d like to try the methods you suggested. May I have your contact information, or could you kindly share the link to that blog? @Jack-Dong-Dong

**Comment by LinLin1031** [comment]
> > Yes, it's also mentioned in Box2Mask, except that repo also just gives a download link for oversegmentation, but it's not clear how this is generated. If you switch to your own dataset, you still don't know how to generate the corresponding file for that pre-processing. Examining the .npy file under the file. There are five values for 'segment', 'per_point_segment_instanceID', 'seg2instanceID', 'per_point_segment_semanticID', 'seg2semanticID', and 'seg2semanticID'. attribute information. The headache is that there is no corresponding xyz coordinates for each point, so this can't be realized in the usual way. @Jack-Dong-Dong
> 
> I found a solution on the blog, and it ran successfully: to generate superpoints index according to the segment_mesh method provided by the segmentator package, you need to pass two parameters, which are the parameters of TriangleMesh, for specific introduction, you can refer to the content related to TriangleMesh in open3d, and TriangleMesh is also generated according to the method provided in open3d, I understand this superpoint as a triangular surface to describe the point cloud features, considering that the data is large, it can be generated based on a single instance and combined together, the above is my approach, I hope to help you. There's an option in the config file to use superpoint, and I think it's possible to change it to false, maybe ignoring this

This method sounds only applicable to mesh data but not to pure point cloud data?

**Comment by Jack-Dong-Dong** [comment]
> Thank you very much for your response; your answer has been very helpful to my research. I’d like to try the methods you suggested. May I have your contact information, or could you kindly share the link to that blog? @Jack-Dong-Dong

Absolutely, it's a pleasure to learn with you, This is my Email:[pengzhen@m.scnu.edu.cn.After](mailto:pengzhen@m.scnu.edu.cn.After) that, we can exchange contact details.

**Comment by Jack-Dong-Dong** [comment]
> > > Yes, it's also mentioned in Box2Mask, except that repo also just gives a download link for oversegmentation, but it's not clear how this is generated. If you switch to your own dataset, you still don't know how to generate the corresponding file for that pre-processing. Examining the .npy file under the file. There are five values for 'segment', 'per_point_segment_instanceID', 'seg2instanceID', 'per_point_segment_semanticID', 'seg2semanticID', and 'seg2semanticID'. attribute information. The headache is that there is no corresponding xyz coordinates for each point, so this can't be realized in the usual way. @Jack-Dong-Dong
> > 
> > 
> > I found a solution on the blog, and it ran successfully: to generate superpoints index according to the segment_mesh method provided by the segmentator package, you need to pass two parameters, which are the parameters of TriangleMesh, for specific introduction, you can refer to the content related to TriangleMesh in open3d, and TriangleMesh is also generated according to the method provided in open3d, I understand this superpoint as a triangular surface to describe the point cloud features, considering that the data is large, it can be generated based on a single instance and combined together, the above is my approach, I hope to help you. There's an option in the config file to use superpoint, and I think it's possible to change it to false, maybe ignoring this
> 
> This method sounds only applicable to mesh data but not to pure point cloud data?

According to the introduction of this article,[https://github.com/loicland/superpoint_graph](url)，Based on the point cloud, he generates a unified polygon, which I understand as a cluster, but abstracts the cluster into a polygon, which is box2mask. The super points are the vertex indexes of these masks, and my method should be to convert the point cloud into a kind of mesh

**Comment by lmyuxian116** [comment]
> Thank you very much for your response; your answer has been very helpful to my research. I’d like to try the methods you suggested. May I have your contact information, or could you kindly share the link to that blog? [@Jack-Dong-Dong](https://github.com/Jack-Dong-Dong)

Hello, could you please share the blog? Thank you very much!

**Comment by Jack-Dong-Dong** [comment]
> > Thank you very much for your response; your answer has been very helpful to my research. I’d like to try the methods you suggested. May I have your contact information, or could you kindly share the link to that blog? [@Jack-Dong-Dong](https://github.com/Jack-Dong-Dong)
> 
> Hello, could you please share the blog? Thank you very much!

This method should come from one of the Segmenter tools, and the generation of SuperPoint requires two mesh parameters, which can be generated with the Open3D package.

**Comment by lmyuxian116** [comment]
> > > Thank you very much for your response; your answer has been very helpful to my research. I’d like to try the methods you suggested. May I have your contact information, or could you kindly share the link to that blog? [@Jack-Dong-Dong](https://github.com/Jack-Dong-Dong)
> > 
> > 
> > Hello, could you please share the blog? Thank you very much!
> 
> This method should come from one of the Segmenter tools, and the generation of SuperPoint requires two mesh parameters, which can be generated with the Open3D package.

How is the performance of the trained model when using superpoints generated by your method?

**Comment by Jack-Dong-Dong** [comment]
> > > > Thank you very much for your response; your answer has been very helpful to my research. I’d like to try the methods you suggested. May I have your contact information, or could you kindly share the link to that blog? [@Jack-Dong-Dong](https://github.com/Jack-Dong-Dong)
> > > 
> > > 
> > > Hello, could you please share the blog? Thank you very much!
> > 
> > 
> > This method should come from one of the Segmenter tools, and the generation of SuperPoint requires two mesh parameters, which can be generated with the Open3D package.
> 
> How is the performance of the trained model when using superpoints generated by your method?

There was no follow-up, and the model was changed due to the large amount of data.
